opendebates/models.py

```diff
@@ -177,8 +197,22 @@ def debate(self):
             self._cached_debate = self.category.debate
         return self._cached_debate
 
+    def save(self, *args, **kwargs):
+        """
+        Update the search_vector field whenever the save() method is called.
+
+        Note: in case it becomes too slow to update the search_vector field (and
+        thus to re-index) when calling the save() method, we can use a different
+        approach to update the search_vector field, such as updating the field
+        for the entire table every few minutes, etc.
+        """
+        super(Submission, self).save(*args, **kwargs)
+        if 'update_fields' not in kwargs or 'search_vector' not in kwargs['update_fields']:
+            self.search_vector = self._search_vectors
+            self.save(update_fields=['search_vector'])
```

Comment:
I think what we might want for this if statement is ```if 'update_fields' not in kwargs or 'search_vector' in kwargs['update_fields']:```. That is, only update the search_vector field if there's no update_fields kwarg (indicating that the user wants to update all fields) or the update_fields kwarg *does* include search_vector (indicating that the user wants to update search_vector).  If the user provides update_fields and excludes search_vector from it, that would seem to indicate they don't want search_vector changed in the database.

Reply:
Alternatively we could use an automatically generated field in Postgres (see docs 12.2.2, https://www.postgresql.org/docs/10/textsearch-tables.html#TEXTSEARCH-TABLES-INDEX). 

I think I'd prefer that to needing to try to do the right thing in .save() always.

Reply:
The comment about the logic for `save()` makes sense, but could you explain what you mean by using an automatically generated field in Postgres? I can't seem to find that in the docs you referenced.

Reply:
Sorry, I was paraphrasing. Let me find the exact thing...

I think I was actually looking at https://www.postgresql.org/docs/12/textsearch-tables.html.  The paragraph starting "Another approach is to create a separate tsvector column".

Generated columns https://www.postgresql.org/docs/12/ddl-generated-columns.html seem to be a new feature in Postgres 12.  We seem to be using Postgres 9 on our hosted Postgres server, and the latest GCP offers is 11, so... I guess we can't do that.

Reply:
Ah, I see now. That would be a better way to do it. Maybe I can add a note in the models file about wanting to use a stored generated column when we can use Postgres12. For now, I guess I'll update the `save()` method